src/main/java/Alvis.java

```diff
@@ -0,0 +1,129 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+
+public class Alvis {
+    private static final String FILE_PATH = "./Save.txt";
+    
+    public static void main(String[] args) throws IOException {
+        Scanner sc = new Scanner(System.in);
+        List<Task> taskList = new ArrayList<>();
+        loadTasks(taskList);
+        System.out.println("So you require my assistance?");
+
+        while (true) {
+            String userInput = sc.nextLine();
+            switch (userInput.split(" ")[0].toLowerCase()) {
+            case "bye":
+                System.out.println("May we meet again");
+                System.exit(0);
+                break;
+            case "list":
+                System.out.println("Let me help you to remember what you need to do ...");
+                for (int i = 0; i < taskList.size(); ++i) {
+                    System.out.println("" + (i + 1) + ". " + taskList.get(i).toString());
+                }
+                break;
+            case "toggle":
+                try {
+                    int index = Integer.parseInt(userInput.substring(7)) - 1;
+                    taskList.get(index).toggleStatus();
+                    System.out.println("As requested, I have changed the status of your task");
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("I cannot do that, the index is invalid");
+                }
+                writeToFile(taskList);
+                break;
+            case "todo":
+                try {
+                    Task toDo = new ToDo(userInput.substring(5));
+                    addTask(toDo, taskList);
+                } catch (StringIndexOutOfBoundsException e) {
+                    System.out.println("Please input a body for the task");
+                }
+                writeToFile(taskList);
+                break;
+            case "delete":
+                try {
+                    int index = Integer.parseInt(userInput.substring(7)) - 1;
+                    taskList.remove(index);
+                    System.out.println("As requested, I have deleted the task");
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("I cannot do that, the index is invalid");
+                }
+                writeToFile(taskList);
+                break;
+            case "deadline":
+                int byIndex = userInput.indexOf("/by");
```

Comment:
I feel rn, the main method is doing everything. Must be tiring for it. 😩 

Suggestion: Instead of main method handling everything like taking input and processing it, you can break those tasks into different classes like ProcessTasks.class, SaveAndLoad.class, etc. according to your taste. 🕺🏼

Reply:
Hahaha indeed my bot was getting tired. I have split up the workload into various other classes. Hopefully, the code is still easy to understand!